server/db.go

```diff
@@ -448,17 +442,15 @@ func (c *Server) fetchIssuerByType(ctx context.Context, issuerType string) (*Iss
 	}
 
 	if c.caches != nil {
-		c.caches["issuer"].SetDefault(issuerType, issuerV3)
+		c.caches["issuer"].SetDefault(issuerType, &issuerV3)
 	}
 
 	return convertedIssuer, nil
 }
 
 func (c *Server) fetchIssuers(issuerType string) (*[]Issuer, error) {
-	if c.caches != nil {
-		if cached, found := c.caches["issuers"].Get(issuerType); found {
-			return cached.(*[]Issuer), nil
-		}
+	if cached := retrieveFromCache(c.caches, "issuers", issuerType); cached != nil {
+		return cached.(*[]Issuer), nil
```

Comment:
Do we need to check here that the type assertion will pass and we have one or more issuers in the slice?

Reply:
To this point, slices are already allowed to be nil w/o needing a pointer. Maybe this is for a different day, but if it's easy enough can you have it not be a pointer to a slice?

Reply:
I thought I got all of these. Good catch. Fixing.